File: backend/internal/models/conversation.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Conversation struct {
	// gorm.Model
	ID              uuid.UUID `gorm:"unique;type:uuid;primaryKey"`
	CreatedAt       time.Time
	UpdatedAt       time.Time
	DeletedAt       *time.Time `gorm:"index"`
	Messages        []Message  `gorm:"foreignKey:ConversationId"`
	BuyerId         uuid.UUID
	Buyer           User `gorm:"foreignKey:BuyerId"`
	SellerId        uuid.UUID
	Seller          User `gorm:"foreignKey:SellerId"`
	TicketListingId uuid.UUID
	TicketListing   TicketListing `gorm:"foreignKey:TicketListingId"`
	Price           float64       `gorm:"not null"`
}

File: frontend/lib/core/models/conversation.dart
import 'package:weezemaster/core/models/ticket_listing.dart';
import 'package:weezemaster/core/models/seller_buyer.dart';

class Conversation {
  final String id;
  final DateTime createdAt;
  final DateTime updatedAt;
  final String buyerId;
  final SellerBuyer buyer;
  final String sellerId;
  final SellerBuyer seller;
  final String ticketListingId;
  final TicketListing ticketListing;
  final List messages;
  final double price;

  Conversation({
    required this.id,
    required this.createdAt,
    required this.updatedAt,
    required this.buyerId,
    required this.buyer,
    required this.sellerId,
    required this.seller,
    required this.ticketListingId,
    required this.ticketListing,
    required this.messages,
    required this.price,
  });

  factory Conversation.fromJson(Map<String, dynamic> json) {
    return Conversation(
      id: json['ID'],
      createdAt: DateTime.parse(json['CreatedAt']),
      updatedAt: DateTime.parse(json['UpdatedAt']),
      buyerId: json['BuyerId'],
      buyer: SellerBuyer.fromJson(json['Buyer']),
      sellerId: json['SellerId'],
      seller: SellerBuyer.fromJson(json['Seller']),
      ticketListingId: json['TicketListingId'],
      ticketListing: TicketListing.fromJson(json['TicketListing']),
      messages: json['Messages'] ?? [],
      price: (json['Price'] ?? 0).toDouble(),
    );
  }
}